File: app/src/main/java/com/onus/demotest/threadpool/CVMainThreadExecutor.java
package com.onus.demotest.threadpool;

import android.os.Handler;
import android.os.Looper;

import java.util.concurrent.Executor;

/**
 * UI线程线程池
 * 
 * @xiandongluo
 */
public class CVMainThreadExecutor implements Executor
{
	private Handler mHandler;

	public CVMainThreadExecutor()
	{
		mHandler = new Handler(Looper.getMainLooper());
	}

	@Override
	public void execute(Runnable command)
	{
		mHandler.post(command);
	}

	public void execute(Runnable command, long delay)
	{
		mHandler.postDelayed(command, delay);
	}

	public void remove(Runnable command)
	{
		mHandler.removeCallbacks(command);
	}
}

File: app/src/main/kotlin/com/onus/demotest/core/threadpool/CVMainThreadExecutor.kt
package com.onus.demotest.core.threadpool

import android.os.Handler
import android.os.Looper
import java.util.concurrent.Executor

open class CVMainThreadExecutor : Executor {
    private val handler = Handler(Looper.getMainLooper())

    override fun execute(command: Runnable) {
        handler.post(command)
    }

    fun execute(command: Runnable, delay: Long) {
        handler.postDelayed(command, delay)
    }

    fun remove(command: Runnable) {
        handler.removeCallbacks(command)
    }
}
